### PR TITLE
Hw02 express

### PR DESCRIPTION
**Pull request by weroszymaniak**
Hej, proszę o sprawdzenie :)

### CODE REVIEW

File: models/contacts.js [hunk 1]
```diff
@@ -1,19 +1,123 @@
...
+    console.log(error.message);
+    throw error;
+  }
+};
 
-const addContact = async (body) => {}
+export const listContacts = async () => {
+  try {
+    const contacts = await fetchContacts();
+    // if (contacts && contacts.length > 0) {
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js]: Please remove commented code in whole application :)

File: models/contacts.js [hunk 2]
```diff
@@ -1,19 +1,123 @@
...
-module.exports = {
-  listContacts,
-  getContactById,
-  removeContact,
-  addContact,
-  updateContact,
-}
+export const getContactById = async (contactId) => {
+  try {
+    const contacts = await fetchContacts();
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js:53]: Do you think its good to fetch contacts before validation?

File: models/contacts.js [hunk 3]
```diff
@@ -1,19 +1,123 @@
...
+    }
+  } catch (error) {
+    console.log("Error adding new contact:", error);
+    throw error;
+  }
+};
+
+export const updateContact = async (contactId, body) => {
+  try {
+    const contacts = await fetchContacts();
```
Comment (AdamChodowiecGoIt91) [inline models/contacts.js:132]: Same here :)

File: routes/api/contacts.js [hunk 4]
```diff
@@ -1,25 +1,149 @@
...
+  }
+
+  const { contactId } = req.params;
+  const { name, email, phone } = req.body;
+
+  if (!name && !email && !phone) {
+    return res.status(400).json({ message: "missing fields" });
+  }
+
+  const updatedFields = {};
```
Comment (AdamChodowiecGoIt91) [inline routes/api/contacts.js]: `updatedFields = { body, email. phone }`

Comment (AdamChodowiecGoIt91) [reply routes/api/contacts.js]: you can simplify it that way

File: models/contacts.js (hunk 1 above)
Comment (weroszymaniak) [reply models/contacts.js]: removed :)

File: models/contacts.js (hunk 2 above)
Comment (weroszymaniak) [reply models/contacts.js:53]: validation added

File: routes/api/contacts.js (hunk 4 above)
Comment (weroszymaniak) [reply routes/api/contacts.js]: changed :)

**Pull request merged**